app/helpers/api/preference.ts

```diff
@@ -40,3 +34,27 @@ export function getTeammateNameDisplaySetting(preferences: PreferenceType[] | Pr
 
     return General.TEAMMATE_NAME_DISPLAY.SHOW_USERNAME;
 }
+
+export function getAdvanceSettingPreferenceAsBool(preferences: Preference[], name: string, defaultValue = false) {
+    return getPreferenceAsBool(preferences, Preferences.CATEGORIES.ADVANCED_SETTINGS, name, defaultValue);
+}
+
+export function getDisplayNamePreferenceAsBool(preferences: Preference[], name: string, defaultValue = false) {
+    return getPreferenceAsBool(preferences, Preferences.CATEGORIES.DISPLAY_SETTINGS, name, defaultValue);
+}
+
+export function getDisplayNamePreference<T>(preferences: Preference[], name: string, defaultValue = '' as unknown) {
+    return getPreferenceValue<T>(preferences, Preferences.CATEGORIES.DISPLAY_SETTINGS, name, defaultValue);
+}
+
+export function getSidebarPreferenceAsBool(preferences: Preference[], name: string, defaultValue = false) {
+    return getPreferenceAsBool(preferences, Preferences.CATEGORIES.SIDEBAR_SETTINGS, name, defaultValue);
+}
+
+export function filterPreferences(preferences?: PreferenceType[]) {
+    if (!preferences?.length) {
+        return preferences;
+    }
+    const categories = new Set(Object.values(CATEGORIES_TO_KEEP));
```

Comment:
Do we need to create this set every time? Does it make sense to have it precalculated  (i.e. outside of the function)?